Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -2350,6 +2352,31 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
         }
         return automaticallyCuratedSymbols
     }
+
+    /// Handles overloaded symbols by grouping them together into one page.
+    private func handleOverloads(with linkResolver: PathHierarchyBasedLinkResolver) {
+        guard FeatureFlags.current.isExperimentalOverloadedSymbolPresentationEnabled else {
+            return
+        }
+        
+        linkResolver.traverseOverloadedSymbols { overloadedSymbolReferences in
+            guard overloadedSymbolReferences.count > 1 else { return }
```

Comment:
Maybe instead of guarding against "groups" containing zero or one overloaded symbols, we could update the logic in `traverseOverloadedSymbols` to guarantee that it only calls the closure for groups with 2 or more elements. In other words, maybe the concept of "overloaded symbols" doesn't make sense when there are none or just one.